src/Site.php

```diff
@@ -188,17 +188,30 @@ public function wp_object(): ?WP_Site
     /**
      * Switches to the blog requested in the request
      *
-     * @param string|integer|null $site_name_or_id
+     * @param string|integer|null $blog_identifier The name or ID of the blog to switch to. If `null`, the current blog.
      * @return integer with the ID of the new blog
      */
-    protected static function switch_to_blog($site_name_or_id)
+    protected static function switch_to_blog($blog_identifier)
     {
-        if ($site_name_or_id === null) {
-            $site_name_or_id = \get_current_blog_id();
+        $current_id = \get_current_blog_id();
+
+        if ($blog_identifier === null) {
+            $blog_identifier = $current_id;
+        }
+
+        $info = \get_blog_details($blog_identifier);
+
+        if (false === $info) {
+            return $current_id;
         }
-        $info = \get_blog_details($site_name_or_id);
-        \switch_to_blog($info->blog_id);
-        return $info->blog_id;
+
+        $blog_identifier = $info->blog_id;
+
+        if ($current_id !== $blog_identifier) {
```

Comment:
This check should be typecast, because `$blog_identifier` might be passed as a string instead of an integer.

```suggestion
        if ($current_id !== (int) $blog_identifier) {
```